spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLocationResolver.java

```diff
@@ -46,80 +41,61 @@
+import static org.springframework.util.ClassUtils.isPresent;
 
 /**
  * @author Ryan Baxter
  */
 public abstract class KubernetesConfigDataLocationResolver
 		implements ConfigDataLocationResolver<KubernetesConfigDataResource>, Ordered {
 
-	static final boolean RETRY_IS_PRESENT = ClassUtils.isPresent("org.springframework.retry.annotation.Retryable",
-			null);
-
-	/**
-	 * Prefix for Config Server imports.
-	 */
-	public static final String PREFIX = "kubernetes:";
+	private static final boolean RETRY_IS_PRESENT = isPresent("org.springframework.retry.annotation.Retryable", null);
 
 	private final Log log;
 
 	public KubernetesConfigDataLocationResolver(DeferredLogFactory factory) {
 		this.log = factory.getLog(KubernetesConfigDataLocationResolver.class);
 	}
 
-	protected String getPrefix() {
-		return PREFIX;
+	protected final String getPrefix() {
+		return "kubernetes:";
 	}
 
 	@Override
-	public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDataLocation location) {
-		if (!location.hasPrefix(getPrefix())) {
-			return false;
-		}
+	public final int getOrder() {
+		return -1;
+	}
 
-		return (CloudPlatform.KUBERNETES.isEnforced(context.getBinder())
-				|| CloudPlatform.KUBERNETES.isDetected(new StandardEnvironment()));
+	@Override
+	public final boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDataLocation location) {
+		return location.hasPrefix(getPrefix())
+				&& (KUBERNETES.isEnforced(context.getBinder()) || KUBERNETES.isDetected(new StandardEnvironment()));
 	}
 
 	@Override
-	public List<KubernetesConfigDataResource> resolve(ConfigDataLocationResolverContext context,
+	public final List<KubernetesConfigDataResource> resolve(ConfigDataLocationResolverContext context,
 			ConfigDataLocation location)
 			throws ConfigDataLocationNotFoundException, ConfigDataResourceNotFoundException {
 		return Collections.emptyList();
 	}
 
 	@Override
-	public List<KubernetesConfigDataResource> resolveProfileSpecific(ConfigDataLocationResolverContext resolverContext,
-			ConfigDataLocation location, Profiles profiles) throws ConfigDataLocationNotFoundException {
-		PropertyHolder propertyHolder = loadProperties(resolverContext);
-		KubernetesClientProperties properties = propertyHolder.kubernetesClientProperties();
+	public final List<KubernetesConfigDataResource> resolveProfileSpecific(
+			ConfigDataLocationResolverContext resolverContext, ConfigDataLocation location, Profiles profiles)
+			throws ConfigDataLocationNotFoundException {
+		PropertyHolder propertyHolder = PropertyHolder.of(resolverContext);
+		KubernetesClientProperties clientProperties = propertyHolder.kubernetesClientProperties();
 		ConfigMapConfigProperties configMapProperties = propertyHolder.configMapConfigProperties();
 		SecretsConfigProperties secretsProperties = propertyHolder.secretsProperties();
 
-		ConfigurableBootstrapContext bootstrapContext = resolverContext.getBootstrapContext();
```

Comment:
this code has moved to `registerProperties`. again, only because its so much easier to follow